pkg/scheduler/api/job_info.go

```diff
@@ -343,17 +343,19 @@ func (ji *JobInfo) FitError() string {
 	return reasonMsg
 }
 
-// ReadyTaskNum returns the number of tasks that are ready.
+// ReadyTaskNum returns the number of tasks that are ready or that is best-effort.
 func (ji *JobInfo) ReadyTaskNum() int32 {
-	occupid := 0
-	for status, tasks := range ji.TaskStatusIndex {
-		if AllocatedStatus(status) ||
-			status == Succeeded {
-			occupid = occupid + len(tasks)
+	var occupied int32
+	for _, task := range ji.Tasks {
+		if task.InitResreq.IsEmpty() {
+			occupied++
+		} else if AllocatedStatus(task.Status) ||
```

Comment:
if so, we're going to go through all pending tasks for this. Use `TaskStatusIndex` to check allocated tasks, and find best-effort pods in pending status.